1.0/docs/devguide/registering-elements.md

```diff
@@ -232,56 +308,104 @@ local DOM when the element is constructed.
-between **parents and light DOM children**. You should not rely on observed
-timing to be identical across browsers, except as noted below.
+    <avatar-list>
+      <my-photo class="photo" src="one.jpg">First photo</my-photo>
+      <my-photo class="photo" src="two.jpg">Second photo</my-photo>
+    </avatar-list>
 
-For a given element:
+`<avatar-list>` is _likely_ to have its `ready` method called before the various
+`<my-photo>` elements do.
 
-*   The `created` callback is always called before `ready`.
-*   The `ready` callback is always called before `attached`.
-*   The `ready` callback is called on any **local DOM children** before it's called
-    on the host element.
+In addition, the user can add light children at any time after
+the parent element has been created. A well-designed element
+should handle having its light DOM manipulated at runtime.
 
-This means that an element's **light DOM children** may be initialized **before or after** 
-the parent element, and an element's **siblings may become `ready` in any order**.
+To avoid timing issues, you can use the following strategies:
 
-For accessing sibling elements when an element initializes, you can call `async` from inside
-the `attached` callback:
+*   Handle light DOM children lazily. For example, a popup menu
+    element may need to count its light DOM children. By counting
+    its `children` when the menu is opened, it can handle the user
+    adding and removing menu items with minimal overhead.
+
+*   To react when children are added and removed, use the
+    [`observeNodes` method](local-dom.html#observe-nodes).
+
+#### Initialization timing for local DOM children
+
+In terms of local DOM and initialization timing, local DOM children are created,
+their property values are set as specified in the template, and `ready` is
+called on them _before_ their parent's `ready` callback is called.
+
+There are two caveats:
+
+ *  `dom-repeat` and `dom-if` templates create DOM **asynchronously**
+    after their properties are updated. For example, if you have a
+    `dom-repeat` in your element's local DOM, the `ready` callback is
+    invoked before the `dom-repeat` finishes creating its instances.
+
+    If you need to know when a `dom-repeat` or `dom-if` creates or
+    removes template instances, listen for its `dom-change` event.
+    See [`dom-change` event](templates.html#dom-change) for details.
+
+*   Polymer guarantees that local DOM children have their `ready` callback called
+    before their parent's; however, it cannot guarantee the same thing for the
+    `attached` callback. This is one fundamental difference between native
+    behavior and  polyfill behavior.
+
+#### Initialization timing for siblings
+
+There are no guarantees with regard to initialization timing between sibling
+elements.
+
+This means that siblings may become `ready` in any order.
+
+For accessing sibling elements when an element initializes, you can call `async`
```

Comment:
There is not. I would like to improve this section with better practices, but I don't have anything ready to go here. I